apps/app/src/app/pages/(components)/components/(autocomplete)/autocomplete--countries.example.ts

```diff
@@ -0,0 +1,61 @@
+import { ChangeDetectionStrategy, Component, computed, signal } from '@angular/core';
+import { HlmAutocomplete } from '@spartan-ng/helm/autocomplete';
+import { provideHlmAutocompleteConfig } from 'libs/helm/autocomplete/src/lib/hlm-autocomplete.token';
+
+type Country = {
+	name: string;
+	code: string;
+	flag: string;
+};
+
+@Component({
+	selector: 'spartan-autocomplete-countries',
+	imports: [HlmAutocomplete],
+	template: `
+		<hlm-autocomplete [filteredOptions]="filteredCountries()" [optionTemplate]="option" [(search)]="search" />
+
+		<ng-template #option let-option>{{ option.flag }} {{ option.name }}</ng-template>
```

Comment:
@ashley-hunter `option` variable is here types as `any`, maybe its because the `HlmAutocomplete<any>` is also inferred as any even though `filteredOptions` is from type `Country[]`. Anything we can help to infer the correct type?

Reply:
Hmm, so the only way this is potentially possible that I am aware of is rather than passing the option template as an input, would be to have it as a child and a structural directive, something like:

```html
<hlm-autocomplete [filteredOptions]="filteredCountries()" [(search)]="search">
	<span *hlmAutocompleteOption="let option">{{ option.flag }} {{ option.name }}</span>
</hlm-autocomplete>
```

You can see how you can implement type guards here:
https://angular.dev/guide/directives/structural-directives#typing-the-directives-context

That being said, in this case we are trying to infer from a generic on the parent component, so I'm not sure if that kind of inference would be possible unfortunately.